[xbmc][windows] Fix a bunch of warnings for windows x64

Paxxi opened the pull request:
<!--- Provide a general summary of your change in the Title above -->
This takes us from ~1260 warnings to ~1030.

Left them as a few separate commits for now for easier review, should be squashed before merge.

this touches on your stuff @tamland  and @garbear so please have a look.
## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

garbear commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/kodi_game_types.h:
```diff
@@ -419,7 +419,7 @@ typedef struct AddonProps_Game
    * Number of proxy DLL paths provided.
    */
-  unsigned int proxy_dll_count;
+  size_t proxy_dll_count;
```
Can you drop these? I've fixed them in a recent commit that I'll PR soon.

BTW, these would require `#include <stdint.h>`

garbear replied:
For reference: https://github.com/garbear/xbmc/commit/25d5bf9

Paxxi replied:
yep I'll drop them.

garbear replied:
Thanks. I'll get that commit in Kodi in the next few weeks.

garbear commented on xbmc/games/addons/GameClientProperties.h:
```diff
@@ -57,13 +57,13 @@ class CGameClientProperties
 
   // Number of proxy DLLs needed to load the game client
-  unsigned int GetProxyDllCount(void) const { return m_proxyDllPaths.size(); }
+  size_t GetProxyDllCount(void) const { return m_proxyDllPaths.size(); }
```
Can you drop these? I fixed this in an upcoming comment by moving the both to the .cpp file.

garbear commented on xbmc/windows/GUIMediaWindow.cpp:
```diff
@@ -1650,28 +1650,28 @@ bool CGUIMediaWindow::OnPopupMenu(int itemIdx)
   auto addonMenuRange = std::make_pair(buttons.size(), buttons.size() + addonMenu.size());
   for (const auto& menu : addonMenu)
-    buttons.emplace_back(-buttons.size(), menu->GetLabel(*item));
+    buttons.emplace_back(buttons.size(), menu->GetLabel(*item));
```
The `-` was probably used to prevent conflicts with other positive-ID buttons added. Probably keep the `-` and cast to int.

Paxxi replied:
yeah it probably was but it's never been working that way, removing the `-` retains the current behaviour and removes the warning so it felt like the safest fix for now.

tamland replied:
Are you sure about that? If you look at CContextButtons::Add there is a dupe check, so if you reuse any numbers in that range, it wont add it.

Paxxi replied:
buttons.size() returns a size_t which is unsigned so the negation does nothing and even if it did work buttons is `std::vector<std::pair<unsigned int, std::string>>` so it would never be negative.

Doing `buttons.emplace_back(-static_cast<int>(buttons.size()), menu->GetLabel(*item));` would work but result in very large index values as they would be cast to unsigned.

Paxxi replied:
I'll do a test to be sure. I found it because VS Warned about unary minus on unsigned operand is still positive

Paxxi replied:
I was wrong it seems. cpp.sh/5qzxz I'll go ahead and fix it

Paxxi replied:
Okay, replaced the negation with the bitwise negation operator instead to provide similar behavior.

Paxxi commented:
ping @tamland

Paxxi committed:
Silence warnings about throw() deprecation

Silence narrow cast warnings in swig modules

Fix a bunch more narrowing warnings
